src/native/libs/System.Globalization.Native/pal_locale.m

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <stdlib.h>
+#include "pal_locale_internal.h"
+
+#import <Foundation/Foundation.h>
+
+const char* DetectDefaultAppleLocaleName()
+{
+    NSLocale *currentLocale = [NSLocale currentLocale];
+    NSString *localeName = @"";
+    
+    if (!currentLocale)
+    {
+        return strdup([localeName UTF8String]);
+    }
+
+    if ([currentLocale.languageCode length] > 0 && [currentLocale.countryCode length] > 0)
+    {
+        localeName = [NSString stringWithFormat:@"%@-%@", currentLocale.languageCode, currentLocale.countryCode];
+    }
+    else
+    {
+        localeName = [currentLocale.localeIdentifier stringByReplacingOccurrencesOfString:@"_" withString:@"-"];
+    }
+    
+    return strdup([localeName UTF8String]);
```

Comment:
When is this freed if the return value is typed as `const`?

Reply:
Good question.  The return type of `uloc_getDefault` is `const char*`.  We would have to somehow know if we are using a value returned from `DetectDefaultAppleLocaleName` in order to free it. 

Perhaps moving the call up to https://github.com/dotnet/runtime/blob/523914d6cdcbce5a2a50404a91cff4fbb978c9b0/src/native/libs/System.Globalization.Native/pal_locale.c#L212 would be better.  Thoughts?